sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +63,30 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFiltersToFileIndex(
```

Comment:
this pushes data filters to the underlying file format as well. How about
```
protected var partitionFilters = Seq.empty[Expression]
protected var dataFilters = Seq.empty[Expression]
protected var pushedDataFilters = Seq.empty[Expression]
...
def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
  this.partitionFilters = partitionFilters
  this.dataFilters = dataFilters
  this.pushedDataFilters = pushDataFilters(dataFilters)
}

protected def pushDataFilters(dataFilters: Seq[Expression]) = Nil
```

Then file source impl can just override `pushDataFilters`

Reply:
Sounds good. Changed.